OpenEdXMobile/src/main/java/org/edx/mobile/player/VideoPlayer.java

```diff
@@ -293,12 +297,17 @@ private MediaSource getMediaSource(String videoUrl) {
         final DataSource.Factory dataSourceFactory = new DefaultHttpDataSource.Factory().setUserAgent(userAgent);
         final MediaItem mediaItem = MediaItem.fromUri(Uri.parse(videoUrl));
         final MediaSource mediaSource;
-
         if (VideoUtil.videoHasFormat(videoUrl, AppConstants.VIDEO_FORMAT_M3U8)) {
-            mediaSource = new HlsMediaSource.Factory(dataSourceFactory)
+            mediaSource = new HlsMediaSource
+                    .Factory(dataSourceFactory)
+                    .createMediaSource(mediaItem);
+        } else if (VideoUtil.videoHasFormat(videoUrl, AppConstants.VIDEO_FORMAT_MP4)) {
+            mediaSource = new ProgressiveMediaSource
+                    .Factory(dataSourceFactory)
```

Comment:
better to check the isValid URL cuz as per the document edx supported multiple formats, so there is the chance to have formats other then the `.mp4`. please consider the [Working with Video Components ](https://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/named-release-cypress/creating_content/create_video.html#supported-video-formats) document.

Reply:
Good Catch. 👍